pkg/operator/sync.go

```diff
@@ -14,6 +21,17 @@ func syncControllers(c serviceCAOperator, operatorConfig *operatorv1.ServiceCA)
 		return err
 	}
 
+	cleanupDeprecated.Lock()
+	if !cleanupDeprecated.done {
+		err := cleanupDeprecatedResources(c)
+		if err != nil {
+			cleanupDeprecated.Unlock()
+			return err
+		}
+		cleanupDeprecated.done = true
+	}
+	cleanupDeprecated.Unlock()
```

Comment:
alternative, less ugly: get an configmap informer for the operator namespace and use its lister to look for the resource unconditionally.

Reply:
Less ugly for this PR maybe, but that seems like huge overhead given the cleanup required by the controller unification PR:

https://github.com/openshift/service-ca-operator/pull/89/commits/d41d2c6e31f60d9a628adfaae4b5b79c66381d76#diff-bd2d316d0e70d82715b2108d2a21adb8R235